src/instructlab/system/info.py

```diff
@@ -146,21 +155,41 @@ def _instructlab_info():
     return {f"{name}.version": ver for name, ver in pkgs}
 
 
-def get_sysinfo() -> typing.Dict[str, typing.Any]:
-    """Get system information"""
-    d = {}
-    d.update(_platform_info())
-    d.update(_instructlab_info())
-    d.update(_torch_info())
-    d.update(_torch_cuda_info())
-    d.update(_torch_hpu_info())
-    d.update(_llama_cpp_info())
-    return d
+def add_to_category(
+    categories: dict[str, list[tuple[str, typing.Any]]], category: str, data: dict
+):
+    """Add data to the correct category"""
+    if category not in categories:
+        categories[category] = []
+    categories[category].extend(data.items())
+
+
+def get_sysinfo() -> typing.Dict[str, list[tuple[str, typing.Any]]]:
+    """Get system information and categorize it directly"""
+    categories: dict[str, list[tuple[str, typing.Any]]] = {}
+    add_to_category(categories, "platform", _platform_info())
+    add_to_category(categories, "instructlab", _instructlab_info())
+    add_to_category(categories, "torch", _torch_info())
+    add_to_category(categories, "torch", _torch_cuda_info())
+    add_to_category(categories, "torch", _torch_hpu_info())
+    add_to_category(categories, "llama_cpp_python", _llama_cpp_info())
+    return categories
+
+
+def categorize_and_print(categories: typing.Dict[str, list[tuple[str, typing.Any]]]):
+    """Print system information categorized by type"""
+    for idx, (category, items) in enumerate(categories.items()):
+        if idx > 0:
+            print()
+        display_name = DISPLAY_NAMES.get(category, category.capitalize())
```

Comment:
You could do away with this now that the categories are inserted in get_sysinfo and instead make the category names be the display names you want.